File: bank-service/src/main/java/rs/edu/raf/service/CardIssuerService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.CardIssuerCreateDTO;
import rs.edu.raf.dto.CardIssuerDTO;

import java.util.List;

public interface CardIssuerService {
    /**
     * Creates card issuer
     * @param cardIssuerCreateDTO DTO which contains information about card issuer
     * @return {@link CardIssuerDTO} object representing created card issuer
     */
    CardIssuerDTO createCardIssuer(CardIssuerCreateDTO cardIssuerCreateDTO);

    /**
     * Gets all card issuers
     * @return List of {@link CardIssuerDTO} representing card issuers
     */
    List<CardIssuerDTO> getAllCardIssuers();

    /**
     * Deletes card issuer
     * @param id the id of the cart issuer to be deleted
     */
    void deleteCardIssuer(Long id);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/CardIssuerService.kt
package org.example.bankservicekotlin.service

import org.example.bankservicekotlin.dto.CardIssuerCreateDTO
import org.example.bankservicekotlin.dto.CardIssuerDTO

interface CardIssuerService {
    /**
     * Creates card issuer
     * @param cardIssuerCreateDTO DTO which contains information about card issuer
     * @return [CardIssuerDTO] object representing created card issuer
     */
    fun createCardIssuer(cardIssuerCreateDTO: CardIssuerCreateDTO): CardIssuerDTO

    /**
     * Gets all card issuers
     * @return List of [CardIssuerDTO] representing card issuers
     */
    fun getAllCardIssuers(): List<CardIssuerDTO>

    /**
     * Deletes card issuer
     * @param id the id of the cart issuer to be deleted
     */
    fun deleteCardIssuer(id: Long)
}